internal/archive/archive.go

```diff
@@ -164,6 +201,15 @@ func openUbuntu(options *Options) (Archive, error) {
 			Dir: options.CacheDir,
 		},
 		pubKeys: options.PubKeys,
+		baseURL: archiveURL(options.Pro, options.Arch),
+	}
+
+	if options.Pro != "" {
+		creds, err := findCredentials(archive.baseURL)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
But non-pro archives will not have credentials though. Hence, we cannot set the credentials directly. Unless, you want to assign `nil` explicitly.

Reply:
I would still prefer that, having `creds := nil` and then the if statement. In general I think it is easier to read if the object is created with all the attributes instead of it being mutated later.

Reply:
Done in dd75c92.